subsys/net/lib/mqtt_sock/mqtt_encoder.c

```diff
@@ -153,7 +153,8 @@ static u8_t packet_length_encode(u32_t length, struct buf_ctx *buf)
 {
 	u8_t encoded_bytes = 0;
 
-	MQTT_TRC(">> length:0x%08x cur:%p, end:%p", length, buf->cur, buf->end);
+	MQTT_TRC(">> length:0x%08x cur:%p, end:%p", length,
+		 (buf == NULL) ? 0 : buf->cur, (buf == NULL) ? 0 : buf->end);
```

Comment:
The checks would look a bit nicer and shorter when done like this
```
buf ? buf->cur : 0, buf ? buf->end : 0);
```

Reply:
> I don't know if you follow or not, but in parallel folks try to make us misra-compliant

I know the effort but want to resist it still. Anyway, the coccinelle script would change these when the PR is accepted.